code/datums/components/nanites.dm

```diff
@@ -66,7 +66,7 @@
 	if(isliving(parent))
 		host_mob = parent
 
-		if(!(host_mob.mob_biotypes & (MOB_ORGANIC|MOB_UNDEAD))) //Shouldn't happen, but this avoids HUD runtimes in case a silicon gets them somehow.
+		if(!(host_mob.mob_biotypes & ((MOB_ORGANIC || MOB_UNDEAD || MOB_ROBOTIC) && MOB_HUMANOID))) //Shouldn't happen, but this avoids HUD runtimes in case a silicon gets them somehow.
```

Comment:
This is NOT how bitfields work. I recommend checking the documentation for bitwise ops in the dm reference.

A proper implementation of this check to work how you *want* it to work would be

```suggestion
		if(!(host_mob.mob_biotypes & (MOB_ORGANIC|MOB_UNDEAD|MOB_ROBOTIC)) || !(host_mob.mob_biotypes & MOB_HUMANOID)) //Shouldn't happen, but this avoids HUD runtimes in case a silicon gets them somehow.
```


Almost identical issues apply for all other instances of bitfield check changes in this PR.